consensus/blake3pow/consensus.go

```diff
@@ -340,6 +339,49 @@ func CalcDifficulty(config *params.ChainConfig, time uint64, parent *types.Heade
 	return x
 }
 
+// CalcDifficultyAtIndex is the difficulty adjustment algorithm. It returns
+// the difficulty that a new block should have when created at time
+// given the parent block's time and difficulty.
+// NOTE: This is essentially the Ethereum DAA, without a 'difficulty bomb'
+func CalcDifficultyAtIndex(config *params.ChainConfig, time uint64, parent *types.Header, context int) *big.Int {
```

Comment:
Rather than duplicate the entire DAA, I'd prefer to see `CalcDifficulty()` call `CalcDifficultyAtIndex()` and just pass in the current index. That way we know there are no differences in the DAA